railties/lib/rails/deprecator.rb

```diff
@@ -2,6 +2,6 @@
 
 module Rails
   def self.deprecator # :nodoc:
-    ActiveSupport::Deprecation.instance
+    @deprecator ||= ActiveSupport::Deprecation.new
```

Comment:
I'm rereading https://github.com/rails/rails/pull/46049/files#r974695808, and now I just remember that maybe this one should be the instance otherwise we can't silence deprecation in the singleton anymore?

Reply:
We can silence deprecations in the singleton by using `ActiveSupport::Deprecation.silence` like it was possible in 7.0. That will cause a deprecation for the use of `silence` though.
In a Rails app, we can use `Rails.application.deprecators.silence` and it will silence the deprecation without also causing a deprecation, because `ActiveSupport.deprecator` (the singleton instance) is in the application deprecators as soon as the application is initialized.

Reply:
`Rails.deprecator` is the deprecator for the `railties` gem, similar to how `ActiveRecord.deprecator` is the deprecator for the `activerecord` gem.  It is currently used in a few places:

https://github.com/rails/rails/blob/a9506eb06243d309472071a5d65a3307d15580d4/railties/lib/rails/application/configuration.rb#L339-L347

https://github.com/rails/rails/blob/2d97136e38080edcb908d4b130b2fdda7c984b34/railties/lib/rails/generators/testing/behavior.rb#L112

https://github.com/rails/rails/blob/8542b748ba3eb257321fe0bc31742c72baf4975d/railties/lib/rails/commands/dbconsole/dbconsole_command.rb#L66-L72

However, we could use `ActiveSupport.deprecator` for those if you prefer.

With or without `Rails.deprecator`, `Rails.application.deprecators.silence` will still silence `ActiveSupport::Deprecation.warn`:

  ```irb
  irb> ActiveSupport::Deprecation.warn "boom"
  DEPRECATION WARNING: boom
  DEPRECATION WARNING: Calling warn on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use your own Deprecation object instead)

  irb> Rails.application.deprecators.silence { ActiveSupport::Deprecation.warn "boom" }
  # No output
  ```

Reply:
If that is the railties deprecation we should rename it in the register to `deprecators[:railties]`, that was the point of my comment in the linked PR.

Reply:
@rafaelfranca Done in #47919.